prime-dotnet-webapi/Services/OrganizationClaimService.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Prime.Models;
+using Prime.ViewModels;
+using Prime.Models.Api;
+
+namespace Prime.Services
+{
+    public class OrganizationClaimService : BaseService, IOrganizationClaimService
+    {
+        private readonly IBusinessEventService _businessEventService;
+
+        public OrganizationClaimService(
+            ApiDbContext context,
+            IHttpContextAccessor httpContext,
+            IBusinessEventService businessEventService)
+            : base(context, httpContext)
+        {
+            _businessEventService = businessEventService;
+        }
+
+        public async Task<bool> HasClaimAsync(int organizationId)
+        {
+            return await _context.OrganizationClaims
+                .AsNoTracking()
+                .AnyAsync(oc => oc.OrganizationId == organizationId);
+        }
+
+        public async Task<bool> DeleteClaimAsync(int claimId)
+        {
+            var claim = await _context.OrganizationClaims
+                .SingleOrDefaultAsync(oc => oc.Id == claimId);
+            if (claim == null)
+            {
+                return false;
+            }
+
+            _context.OrganizationClaims.Remove(claim);
+            int numAffected = await _context.SaveChangesAsync();
+            return numAffected == 1;
+        }
+
+        public Task<OrganizationClaim> GetOrganizationClaimAsync(int claimId)
+        {
+            return _context.OrganizationClaims
+                .SingleOrDefaultAsync(oc => oc.Id == claimId);
+        }
+
+        public Task<OrganizationClaim> GetOrganizationClaimByOrgIdAsync(int organizationId)
```

Comment:
These two methods are not marked `async`, and are instead returning an awaitable Task rather than awaiting the query and returning the value. There is a really interesting discussion to be had here on the nuance of each; was this done on purpose or was it an oversight?

Reply:
Yeah, let's talk a bit later today  :-)